conda/models/records.py

```diff
@@ -370,6 +370,8 @@ def record_id(self):
         #              channel_name/subdir:namespace:name-version-build_number-build_string
         return "%s/%s::%s-%s-%s" % (self.channel.name, self.subdir,
                                     self.name, self.version, self.build)
+    def to_query_param(self):
+        return self.name
```

Comment:
the method signature of `SubdirData.queryAll` notes `package_ref_or_match_spec`, which I believe references `PackageRef` a legacy representation of `PackageRecord`. So, I believe this is necessary though I don't see any instance in the current code base that passes in `PackageRecord` to `queryAll`.